packages/calcite-components/src/components/swatch/swatch.tsx

```diff
@@ -0,0 +1,483 @@
+import { PropertyValues } from "lit";
+import { createRef } from "lit-html/directives/ref.js";
+import {
+  LitElement,
+  property,
+  createEvent,
+  h,
+  method,
+  state,
+  JsxNode,
+  Fragment,
+} from "@arcgis/lumina";
+import Color, { ColorInstance } from "color";
+import { getModeName, slotChangeHasAssignedElement } from "../../utils/dom";
+import { Scale, SelectionMode } from "../interfaces";
+import { componentFocusable } from "../../utils/component";
+import {
+  InteractiveComponent,
+  InteractiveContainer,
+  updateHostInteraction,
+} from "../../utils/interactive";
+import type { SwatchGroup } from "../swatch-group/swatch-group";
+import { hexify } from "../color-picker/utils";
+import { COLORS, CHECKER_DIMENSIONS } from "../color-picker-swatch/resources";
+import { CSS, SLOTS, IDS } from "./resources";
+import { styles } from "./swatch.scss";
+
+declare global {
+  interface DeclareElements {
+    "calcite-swatch": Swatch;
+  }
+}
+
+export class Swatch extends LitElement implements InteractiveComponent {
+  //#region Static Members
+
+  static override styles = styles;
+
+  //#endregion
+
+  //#region Private Properties
+
+  private internalColor: ColorInstance;
+
+  private containerEl = createRef<HTMLDivElement>();
+
+  //#endregion
+
+  //#region State Properties
+
+  @state() private hasImage = false;
+
+  //#endregion
+
+  //#region Public Properties
+
+  /**
+   * Specifies the component's color
+   *
+   * @see https://developer.mozilla.org/en-US/docs/Web/CSS/color_value
+   */
+  @property() color: string | null;
+
+  /** When `true`, interaction is prevented and the component is displayed with lower opacity. */
+  @property({ reflect: true }) disabled = false;
+
+  /**
+   * When true, enables the swatch to be focused, and allows the `calciteSwatchSelect` to emit.
+   * This is set to `true` by a parent Swatch Group component.
+   *
+   * @private
+   */
+  @property() interactive = false;
+
+  /**
+   * Accessible name for the component.
+   *
+   * @required
+   */
+  @property() label: string;
+
+  /** @private */
+  @property() parentSwatchGroup: SwatchGroup["el"];
+
+  /** Specifies the size of the component. When contained in a parent `calcite-swatch-group` inherits the parent's `scale` value. */
+  @property({ reflect: true }) scale: Scale = "m";
+
+  /** When `true`, the component is selected. */
+  @property({ reflect: true }) selected = false;
+
+  /**
+   * This internal property, managed by a containing `calcite-swatch-group`, is
+   * conditionally set based on the `selectionMode` of the parent
+   *
+   * @private
+   */
+  @property() selectionMode: Extract<
+    "multiple" | "single" | "single-persist" | "none",
+    SelectionMode
+  > = "none";
+
+  /** The component's value. */
+  @property() value: any;
```

Comment:
should we type this to a string and reflect it?

Reply:
We seem to be pretty inconsistent here across components that have a `value` prop - many use the above `any`, some use `string`, etc, only a couple seem to be reflected. Maybe something to look at across all of these instances of `value`?

Reply:
Should I leave this as-is for now and address when we look across other components?

Reply:
Yep sounds good. This is a broader issue.